rusoto/credential/src/secrets.rs

```diff
@@ -0,0 +1,59 @@
+/// Newtype (pattern) to protect secret credentials stored as Strings.
+#[derive(Clone)]
+pub struct Secret(String);
+
+impl From<String> for Secret {
+    fn from(s: String) -> Self {
+        Secret(s)
+    }
+}
+
+/// Allow dereferencing Secrets as &str.
+///
+/// ```rust
+/// # use rusoto_credential::Secret;
+/// assert_eq!(Secret::from("hello".to_string()).as_ref(), "hello");
+/// ```
+impl AsRef<str> for Secret {
+    fn as_ref(&self) -> &str {
+        return self.0.as_str();
+    }
+}
+
+impl Drop for Secret {
+    fn drop(&mut self) {
+        //TODO Should we use (and depend on) https://docs.rs/zeroize/1.0.0/zeroize/ instead?
```

Comment:
I would prefer to use a published crate for doing any `unsafe` work, as I'm 100% uneducated on `unsafe` best practices. 😆

Reply:
I agree, I will use zeroize instead.